plugins/irma-client/state-client.js

```diff
@@ -0,0 +1,368 @@
+    if (this._options.debugging) {
+      const payloadError = payload ? `with payload ${payload}` : '';
+      console.error(
+        `Unknown transition, tried transition ${transition}`,
+        payloadError
+      );
+    }
+    return false;
+  }
+
+  _updatePairingState(prevTransition, continueOnSecondDevice) {
+    return Promise.resolve()
+      .then(() => {
+        if (!this._options.state.pairing) return Promise.resolve();
+
+        // onlyEnableIf may return 'undefined', so we force conversion to boolean by doing a double negation (!!).
+        const shouldBeEnabled =
+          continueOnSecondDevice &&
+          !!this._options.state.pairing.onlyEnableIf(this._mappings);
+
+        // Skip the request when the pairing method is correctly set already.
+        if (shouldBeEnabled === this._pairingEnabled) return Promise.resolve();
+
+        if (!this._mappings.frontendAuth) {
+          console.warn(
+            'Pairing cannot be enabled, because no frontendAuth token is provided. This might be a security risk.'
+          );
+          return Promise.resolve();
+        }
+        this._pairingEnabled = shouldBeEnabled;
+
+        // If pairing should be enabled, parse the pairing options struct.
+        const options = shouldBeEnabled
+          ? { pairingMethod: this._options.state.pairing.pairingMethod }
+          : { pairingMethod: 'none' };
+        return this._updateFrontendOptions(options);
+      })
+      .then(() =>
+        this._stateMachine.selectTransition(({ validTransitions }) => {
+          if (continueOnSecondDevice) {
+            return validTransitions.includes('showQRCode')
+              ? {
+                  transition: 'showQRCode',
+                  payload: {
+                    qr: JSON.stringify(this._mappings.sessionPtr),
+                    showBackButton: prevTransition === 'chooseQR',
+                  },
+                }
+              : false;
+          } else {
+            return validTransitions.includes('showIrmaButton')
+              ? {
+                  transition: 'showIrmaButton',
+                  payload: {
+                    mobile: this._getMobileUrl(this._mappings.sessionPtr),
+                  },
+                }
+              : false;
+          }
+        })
+      )
+      .catch((err) => {
+        if (this._options.debugging)
+          console.error('Error received while updating pairing state:', err);
+        this._handleNoSuccess('fail', err);
+      });
+  }
+
+  _pairingCompleted() {
+    const delay = new Promise((resolve) => {
+      setTimeout(resolve, this._options.state.pairing.minCheckingDelay);
+    });
+    const url = this._options.state.pairing.completedUrl(this._mappings);
+
+    // eslint-disable-next-line compat/compat
+    return fetch(url, {
+      method: 'POST',
+      headers: { Authorization: this._mappings.frontendAuth },
+    })
+      .finally(() => delay)
```

Comment:
This `delay` afterwards does not work, since the request is already send then. The change in state is not handled by this function, but by `_serverStateChange`. This means that this line is useless in the happy flow now. Especially when using `sse` is gives a bit of a weird user experience now. The pairing state disappears sooner than a user can look up from his phone and look to the desktop screen again.